src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/Diagnostics/IncrementalBuildFailureDetector.cs

```diff
@@ -114,11 +112,9 @@ int IVsUpdateSolutionEvents2.UpdateProjectCfg_Done(IVsHierarchy pHierProj, IVsCf
                     return HResult.OK;
                 }
 
-                if (IsRelevantBuild(dwAction))
+                if (IsRelevantBuild(dwAction) && pCfgProj is IVsBrowseObjectContext { ConfiguredProject: { } configuredProject })
                 {
-                    UnconfiguredProject? unconfiguredProject = _projectServiceAccessor.GetProjectService().GetUnconfiguredProject(pHierProj, appliesToExpression: BuildUpToDateCheck.AppliesToExpression);
-
-                    IProjectChecker? checker = unconfiguredProject?.Services.ExportProvider.GetExportedValueOrDefault<IProjectChecker>();
+                    IProjectChecker? checker = configuredProject.Services.ExportProvider.GetExportedValueOrDefault<IProjectChecker>();
```

Comment:
This is the core change. Here we are using the `pCfgProj` to obtain the relevant `IProjectChecker`. Before we were using the `pHierProj`, which only gave us `UnconfiguredProject`. With the change we can find the `ConfiguredProject` that was built.

The rest of this PR is moving `IProjectChecker` from `UnconfiguredProject` scope to `ConfiguredProject` scope.